### ISSUE TITLE
/node_modules/react-native-code-push/android/codepush.gradle' line: 63  * What went wrong: Could not find matching constructor for: java.io.File(File)

### ISSUE THREAD

**Issue by hamdij0maa** [microsoft/react-native-code-push#2648]
Thanks so much for filing an issue or feature request! Please fill out the following (wherever relevant):

### Steps to Reproduce
1. install react-native-code-push v8.2.0
2.cd android &&  ./gradlew clean

### Actual Behavior
/node_modules/react-native-code-push/android/codepush.gradle' line: 63

* react-native-code-push version: 8.2.0

**Comment by hamdij0maa** [comment]
@caiodeambrosio  https://github.com/microsoft/react-native-code-push/pull/2649

**Comment by dishantwalia** [comment]
getting same error not compatible with AGP 8.0

**Comment by ryskin** [comment]
same (

**Comment by sontruongna** [comment]
same

**Comment by djindal61** [comment]
I am also facing the same issue to upgrade the react-native version from 0.70.7 to 0.72.4 in android and also upgrade the react-native-code-push 8.1.1

**Comment by tykoth** [comment]
Same here, using react-native 0.72.6 and react-native-code-push ^8.1.1

**Comment by nhhai1605** [comment]
Same here with react-native-code-push ^8.1.1

**Comment by milon27** [comment]
same here, any way to solve it

**Comment by imas234** [comment]
same issue, hope the PR gets looked at

**Comment by iita71737** [comment]
Same issue when I update to 8.2.0 , too

**Comment by tykoth** [comment]
i've commented the lines 63 and 64 of the file /node_modules/react-native-code-push/android/codepush.gradle and it works on my local setup but when i run in azure or appcenter throws the error again.

**Comment by kelokchan** [comment]
Here's the temporary patch file using `patch-package` in case everyone needs it

`react-native-code-push+8.2.0.patch`

```
diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
index f701564..4faf49f 100644
--- a/node_modules/react-native-code-push/android/codepush.gradle
+++ b/node_modules/react-native-code-push/android/codepush.gradle
@@ -60,8 +60,8 @@ gradle.projectsEvaluated {
             jsBundleDir = reactBundleTask.property('jsBundleDir').asFile.get()
             resourcesDir = reactBundleTask.property('resourcesDir').asFile.get()
             
-            new File(jsBundleDir).mkdirs()
-            new File(resourcesDir).mkdirs()
+            jsBundleDir.mkdirs()
+            resourcesDir.mkdirs()
 
             jsBundleFile = file("$jsBundleDir/$bundleAssetName")
```

**Comment by tykoth** [comment]
> Here's the temporary patch file using `patch-package` in case everyone needs it
> 
> `react-native-code-push+8.2.0.patch`
> 
> ```
> diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
> index f701564..4faf49f 100644
> --- a/node_modules/react-native-code-push/android/codepush.gradle
> +++ b/node_modules/react-native-code-push/android/codepush.gradle
> @@ -60,8 +60,8 @@ gradle.projectsEvaluated {
>              jsBundleDir = reactBundleTask.property('jsBundleDir').asFile.get()
>              resourcesDir = reactBundleTask.property('resourcesDir').asFile.get()
>              
> -            new File(jsBundleDir).mkdirs()
> -            new File(resourcesDir).mkdirs()
> +            jsBundleDir.mkdirs()
> +            resourcesDir.mkdirs()
>  
>              jsBundleFile = file("$jsBundleDir/$bundleAssetName")
> ```

Your solution works for me!

**Comment by mgscreativa** [comment]
> Here's the temporary patch file using `patch-package` in case everyone needs it
> 
> `react-native-code-push+8.2.0.patch`
> 
> ```
> diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
> index f701564..4faf49f 100644
> --- a/node_modules/react-native-code-push/android/codepush.gradle
> +++ b/node_modules/react-native-code-push/android/codepush.gradle
> @@ -60,8 +60,8 @@ gradle.projectsEvaluated {
>              jsBundleDir = reactBundleTask.property('jsBundleDir').asFile.get()
>              resourcesDir = reactBundleTask.property('resourcesDir').asFile.get()
>              
> -            new File(jsBundleDir).mkdirs()
> -            new File(resourcesDir).mkdirs()
> +            jsBundleDir.mkdirs()
> +            resourcesDir.mkdirs()
>  
>              jsBundleFile = file("$jsBundleDir/$bundleAssetName")
> ```

works! hope the PR gets approved and merged soon

**Comment by FaithfulAudio** [comment]
I am also experiencing this issue. Thanks for the patch!

**Comment by rdnsan** [comment]
> Here's the temporary patch file using `patch-package` in case everyone needs it
> 
> `react-native-code-push+8.2.0.patch`
> 
> ```
> diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
> index f701564..4faf49f 100644
> --- a/node_modules/react-native-code-push/android/codepush.gradle
> +++ b/node_modules/react-native-code-push/android/codepush.gradle
> @@ -60,8 +60,8 @@ gradle.projectsEvaluated {
>              jsBundleDir = reactBundleTask.property('jsBundleDir').asFile.get()
>              resourcesDir = reactBundleTask.property('resourcesDir').asFile.get()
>              
> -            new File(jsBundleDir).mkdirs()
> -            new File(resourcesDir).mkdirs()
> +            jsBundleDir.mkdirs()
> +            resourcesDir.mkdirs()
>  
>              jsBundleFile = file("$jsBundleDir/$bundleAssetName")
> ```

works for me thanks!

**Comment by tuiza** [comment]
> Here's the temporary patch file using `patch-package` in case everyone needs it
> 
> `react-native-code-push+8.2.0.patch`
> 
> ```
> diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
> index f701564..4faf49f 100644
> --- a/node_modules/react-native-code-push/android/codepush.gradle
> +++ b/node_modules/react-native-code-push/android/codepush.gradle
> @@ -60,8 +60,8 @@ gradle.projectsEvaluated {
>              jsBundleDir = reactBundleTask.property('jsBundleDir').asFile.get()
>              resourcesDir = reactBundleTask.property('resourcesDir').asFile.get()
>              
> -            new File(jsBundleDir).mkdirs()
> -            new File(resourcesDir).mkdirs()
> +            jsBundleDir.mkdirs()
> +            resourcesDir.mkdirs()
>  
>              jsBundleFile = file("$jsBundleDir/$bundleAssetName")
> ```

I'm using RN 0.73.2 and this solution worked for me, thanks 😊

**Comment by 3mv3** [comment]
> Here's the temporary patch file using `patch-package` in case everyone needs it
> 
> `react-native-code-push+8.2.0.patch`
> 
> ```
> diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
> index f701564..4faf49f 100644
> --- a/node_modules/react-native-code-push/android/codepush.gradle
> +++ b/node_modules/react-native-code-push/android/codepush.gradle
> @@ -60,8 +60,8 @@ gradle.projectsEvaluated {
>              jsBundleDir = reactBundleTask.property('jsBundleDir').asFile.get()
>              resourcesDir = reactBundleTask.property('resourcesDir').asFile.get()
>              
> -            new File(jsBundleDir).mkdirs()
> -            new File(resourcesDir).mkdirs()
> +            jsBundleDir.mkdirs()
> +            resourcesDir.mkdirs()
>  
>              jsBundleFile = file("$jsBundleDir/$bundleAssetName")
> ```

For anyone patching manually using yarn berry + `yarn patch react-native-code-push` on windows, the patch file may not be generated properly and will have extra diffs for `a/windows-legacy/CodePush.Net46`, these need deleting from the patch file. It needs to look exactly as above (although I needed an extra carriage return for it to work) 🤷‍♂️

**Comment by AnatolyPristensky** [comment]
The fix has been merged, reopened issue for now. Will close it once release with the fix available.

**Comment by AnatolyPristensky** [comment]
The fix available in the latest [8.2.1 release.](https://github.com/microsoft/react-native-code-push/releases/tag/v8.2.1) 
Closing this issue.
